### DISCUSSION TITLE
Distribution of weighting | Comprehension problem

### DISCUSSION THREAD

**Discussion by GregorWedlich** [nftchef/art-engine#140] [Help + Q&A] [answered]
Hello,

I hope I'm not the 100th user to ask this now. But how does it work with the weighting? Can someone explain? Haven't found anything useful or didn't understand :(

How do I specify the distribution if I want only 1% of 100 NFTs to get the "Legend" trait? How do I calculate the weighting so that, for example, for "Common" I distribute the Axe.png at 15% and the Rollers at 14%?

My layer structure:

```
Accessory:
  Common // I want the axe to get 15% and so on.
    Axe#15.png 
    None#17.png
    Pitchfork#16.png
    Roller#14.png
    Shovel#16.png
  Rare
    Trident#5.png
    Sledge#5.png
    Staff#5.png
  Epic
    Pencil#3.png
  Legend
    Asclepius#1.png
 ```
 
 So if anyone can tell me what formula I'm using to calculate this or what I'm doing wrong in my folder structure, that would be awesome!
 
 Thanks in advance :)

**Comment by Symyon** [comment, marked as answer]
TL;DR if you want to use percentages and don't like weights, make sure the sum of all your weights is 100, and then each trait weight will be a percentage.

Weights are similar to percentages, only on a total that is not limited to 100. Percent - literally means over 100, so 1% is 1/100. With weights is the same just the "100" now is any number wich is total of your weights. 
So if you have trait1 having weight of 3 and if you sum up all other weights you get a number of 150, that means that your trait1 will have the probability of 3/150 to be generated, which is the same as 2/100, or 2%.

**Reply by GregorWedlich** [reply]
Ok thanks for your detailed explanation!

One question: In the first part of your answer, you mean that I should not use nested folders? But the weighting of all characteristics must total 100? Right?

```
Accessory:
    Axe#15.png 
    None#17.png
    Pitchfork#16.png
    Roller#14.png
    Shovel#16.png
    Trident#5.png
    Sledge#5.png
    Staff#5.png
    Pencil#3.png
    Lightsaber#3.png
    Asclepius#1.png
```
